### PR TITLE
fix: insert and delete executor

### PR DESCRIPTION
**Pull request by aryan-rajoria**
Fixing issues in insert and delete executor based on feedback from @gaurav274 

- [x] Using list comprehension in `insert_executor`.
- [x] Removing `_roll_back` from the executors.
- [x] Using descriptive variable names.
- [x] Applying predicate function in `delete_executor` using the SQLAlchemy.
- [x] Move `single_predicate_to_filter()` to ExpressionUtils.
- [ ] Make it more robust (take into consideration `AND` and `OR`)
- [ ] Adding test cases for `single_predicate()`.

### CODE REVIEW

**Comment by aryan-rajoria** [comment]
#594 fixes

File: eva/storage/sqlite_storage_engine.py [hunk 1]
```diff
@@ -196,20 +220,12 @@ def delete(self, table: TableCatalogEntry, where_clause: Dict[str, Any]):
...
             # https://stackoverflow.com/questions/34026210/where-filter-from-table-object-using-a-dictionary-or-kwargs
-            for column, value in where_clause.items():
-                if column not in table_columns:
-                    raise Exception(
-                        f"where_clause contains a column {column} not in the table {table_to_delete_from}"
-                    )
-                filter_clause.append(table_to_delete_from.columns[column] == value)
 
             d = table_to_delete_from.delete().where(and_(*filter_clause))
             self._sql_engine.execute(d)
```
Comment (gaurav274) [inline eva/storage/sqlite_storage_engine.py:231]: return the rowcount - `result.rowcount`

File: eva/storage/sqlite_storage_engine.py [hunk 2]
```diff
@@ -184,7 +186,29 @@ def read(
             logger.exception(err_msg)
             raise Exception(err_msg)
 
-    def delete(self, table: TableCatalogEntry, where_clause: Dict[str, Any]):
+    def single_predicate_node_to_filter(
```
Comment (gaurav274) [inline eva/storage/sqlite_storage_engine.py:189]: We can move this to expression utils.

Comment (gaurav274) [reply eva/storage/sqlite_storage_engine.py:189]: And make it more robust. Add test cases for this functions

**Comment by jarulraj** [comment]
#611 takes care of this

**Pull request closed without merging**